x/staking/keeper/msg_server.go

```diff
@@ -257,10 +274,25 @@ func (k msgServer) Delegate(goCtx context.Context, msg *types.MsgDelegate) (*typ
 // BeginRedelegate defines a method for performing a redelegation of coins from a delegator and source validator to a destination validator
 func (k msgServer) BeginRedelegate(goCtx context.Context, msg *types.MsgBeginRedelegate) (*types.MsgBeginRedelegateResponse, error) {
 	ctx := sdk.UnwrapSDKContext(goCtx)
+
 	valSrcAddr, err := sdk.ValAddressFromBech32(msg.ValidatorSrcAddress)
 	if err != nil {
 		return nil, err
 	}
+	valDstAddr, err := sdk.ValAddressFromBech32(msg.ValidatorDstAddress)
+	if err != nil {
+		return nil, err
+	}
+
+	srcValidator, found := k.GetLiquidValidator(ctx, valSrcAddr)
```

Comment:
I was wondering the same thing - would love to hear @zmanian or @faddat's take on it

Imo, when this gets merged into the SDK, we should just have one function here (same with `GetDelegation`), but I assumed it was setup this way intentionally - maybe because other modules call `GetValidator` so we need to keep the legacy function?

Here's the [commit](https://github.com/iqlusioninc/liquidity-staking-module/commit/dc4e1ac50f83cafb3fcd03b74fb0ca10737385dc) - I assume there was good reason for it

Reply:
I think that this could break ibc...  @vuong177 had made that as part of the splice.  Let's check with him.